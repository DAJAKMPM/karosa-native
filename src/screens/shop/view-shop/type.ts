export type ActiveLink = "Popular" | "Latest" | "Price";
