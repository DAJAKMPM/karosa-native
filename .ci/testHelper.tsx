import React, { JSXElementConstructor } from "react";
import { View } from "react-native";
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from "@testing-library/react-native";
import { Formik, FormikValues } from "formik";

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };

  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || "wrapper");
  return { wrapper, ...renderApi };
};

export const renderWithFormikWrapper = <T extends FormikValues>(
  initialValues: T,
  children: React.ReactElement<any, string | JSXElementConstructor<any>>
) => {
  return renderWithWrapper(
    <Formik
      initialValues={initialValues}
      onSubmit={() => console.log("submit")}
      component={() => children}
    />
  );
};
