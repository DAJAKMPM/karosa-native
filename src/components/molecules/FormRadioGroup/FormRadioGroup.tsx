/**
 *
 * FormRadioGroup
 * @format
 *
 */

import React, { FC, Fragment, useState, useCallback } from "react";
import { View } from "react-native";
import { ListItem } from "react-native-elements";
import { useField, useFormikContext } from "formik";
import { useUpdateEffect } from "@app/hooks";
import Radio from "@app/atoms/Radio";

import type { PropsType } from "./types";
import FormRadioGroupStyles from "./styles";

const FormRadioGroup: FC<PropsType> = (props) => {
  const { name, options } = props;
  const [, meta, helpers] = useField(name);
  const { validateOnChange } = useFormikContext();

  const [check, setCheck] = useState({
    value: meta.value || meta.initialValue,
  });

  useUpdateEffect(() => {
    setCheck({
      value: meta.value,
    });
  }, [meta.value]);

  const handlePress = useCallback(
    (val) => () => {
      setCheck({ value: val });

      helpers.setValue(val);

      if (validateOnChange) {
        helpers.setTouched(true);
      }
    },
    [helpers, validateOnChange]
  );

  return (
    <Fragment>
      {options.map((option, index) => (
        <Fragment key={index}>
          <ListItem style={FormRadioGroupStyles.container} bottomDivider>
            <View style={FormRadioGroupStyles.radioContainer}>
              <Radio
                checked={check.value === option.id}
                title={option.label}
                onPress={handlePress(option.id)}
              />
            </View>
          </ListItem>
        </Fragment>
      ))}
    </Fragment>
  );
};

export default FormRadioGroup;
