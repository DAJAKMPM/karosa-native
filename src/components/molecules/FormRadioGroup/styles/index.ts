/**
 *
 * @format
 *
 */

import { StyleSheet } from "react-native";
import { DIMENS } from "@app/styles";

const FormRadioGroupStyles = StyleSheet.create({
  container: {
    width: "95%",
    marginLeft: "2%",
  },
  radioContainer: {
    position: "relative",
    left: "-8.5%",
    flexDirection: "row",
    height: DIMENS.screenWidth * 0.08,
    justifyContent: "flex-start",
    alignItems: "center",
  },
});

export default FormRadioGroupStyles;
