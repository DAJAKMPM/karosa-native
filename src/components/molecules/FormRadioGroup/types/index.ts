/**
 *
 * @format
 *
 */

type Option = {
  id: string;
  label: string;
};

export type PropsType = {
  name: string;
  options: Option[];
};
