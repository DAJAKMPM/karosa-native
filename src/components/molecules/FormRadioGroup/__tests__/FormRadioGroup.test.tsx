import React from "react";
import { renderWithFormikWrapper } from "@app/ci/testHelper";
import type { Option } from "../types";
import FormRadioGroup from "../FormRadioGroup";

const testData: Option[] = [
  {
    id: "1",
    label: "random",
  },
];

describe("FormRadioGroup Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithFormikWrapper(
      { test: "" },
      <FormRadioGroup name="test" options={testData} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
