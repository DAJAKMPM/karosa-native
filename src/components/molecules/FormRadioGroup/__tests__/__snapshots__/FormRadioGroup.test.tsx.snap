// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormRadioGroup Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    style={
      Object {
        "width": "95%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderBottomWidth": 0.5,
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "padding": 14,
        }
      }
      testID="padView"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "height": 30,
            "justifyContent": "space-between",
            "left": "-5.5%",
            "position": "relative",
          }
        }
      >
        <View
          accessibilityRole="checkbox"
          accessibilityState={
            Object {
              "checked": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
          testID="checkbox"
        >
          <View>
            <SvgMock
              height={24}
              width={24}
            />
          </View>
          <Text
            style={
              Object {
                "color": "#242424",
                "fontFamily": "SFProText-Regular",
                "fontSize": 16,
                "fontWeight": "400",
                "lineHeight": 18.75,
                "marginLeft": 8,
              }
            }
            testID="checkboxTitle"
          >
            random
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
